File: src/main/java/com/technews/aggregate/subscribe/domain/Subscribe.java
package com.technews.aggregate.subscribe.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "subscribe")
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Subscribe {
    private String id;
    private String email;
    private Boolean subscribe;
    private String startDt;
    private String endDt;
}

File: src/main/kotlin/com/technews/aggregate/subscribe/domain/Subscribe.kt
package com.technews.aggregate.subscribe.domain

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime

@Document(collection = "subscribe")
class Subscribe(
    @Id
    val id: String? = null,
    val email: String = "",
    var subscribe: Boolean = false,
    val startDt: LocalDateTime = LocalDateTime.now(),
    var endDt: LocalDateTime = LocalDateTime.now(),
) {
    fun unsubscribe() {
        subscribe = false
        endDt = LocalDateTime.now()
    }
}
